src/syntax/translation/desugar/mod.rs

```diff
@@ -148,6 +148,64 @@ fn desugar_lam(
         })
 }
 
+fn desugar_bindings(env: &mut DesugarEnv, items: &[concrete::Item]) -> Vec<raw::Item> {
+    items.iter().map(|concrete_item| match *concrete_item {
+        concrete::Item::Declaration {
+            name: (start, ref name),
+            ref ann,
+        } => {
+            let term = ann.desugar(&env);
+
+            raw::Item::Declaration {
+                label_span: ByteSpan::from_offset(start, ByteOffset::from_str(name)),
+                label: Label(name.clone()),
+                binder: env.on_item(name),
+                term,
+            }
+        },
+        concrete::Item::Definition {
+            name: (start, ref name),
+            ref params,
+            ref return_ann,
+            ref body,
+        } => {
+            let return_ann = return_ann.as_ref().map(<_>::as_ref);
+            let term = desugar_lam(&env, params, return_ann, body);
+
+            raw::Item::Definition {
+                label_span: ByteSpan::from_offset(start, ByteOffset::from_str(name)),
+                label: Label(name.clone()),
+                binder: env.on_item(name),
+                term,
+            }
+        },
+        concrete::Item::Error(_) => unimplemented!("error recovery"),
+    }).collect()
+}
+
+fn desugar_let(
+    env: &DesugarEnv,
+    items: &[concrete::Item],
+    body: &concrete::Term,
+) -> raw::RcTerm {
+    //unimplemented!("let bindings");
+    let mut env = env.clone();
+    let items = desugar_bindings(&mut env, items);
+    let body = body.desugar(&env);
+
+    let hole = raw::RcTerm::from(raw::Term::Hole(ByteSpan::default()));
+
+    items.into_iter().rev().fold(body, |acc, item| match item {
+        raw::Item::Declaration{..} => acc, // TODO Let declarations
```

Comment:
Super obsessive nit-pick: I try to use a colon in TODOs, eg: `// TODO: let declarations`

Reply:
We might want to reuse the module level desugaring for this, but I'm cool with keeping this PR simpler.